File: app/src/main/java/com/besome/sketch/fragments/projects_store/adapters/StorePagerProjectsAdapter.java
package com.besome.sketch.fragments.projects_store.adapters;

import static mod.ilyasse.utils.UI.loadImageFromUrl;

import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.besome.sketch.fragments.projects_store.api.ProjectModel;
import com.sketchware.remod.databinding.ViewStoreProjectPagerItemBinding;

import java.util.List;

public class StorePagerProjectsAdapter extends RecyclerView.Adapter<StorePagerProjectsAdapter.ViewHolder> {

    private final List<ProjectModel.Project> projects;

    public StorePagerProjectsAdapter(List<ProjectModel.Project> projects) {
        this.projects = projects;
    }

    @NonNull
    @Override
    public StorePagerProjectsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        ViewStoreProjectPagerItemBinding binding = ViewStoreProjectPagerItemBinding.inflate(inflater, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(StorePagerProjectsAdapter.ViewHolder holder, int position) {
        ProjectModel.Project project = projects.get(position);

        holder.binding.projectTitle.setText(project.getTitle());
        holder.binding.projectDesc.setText(project.getDescription());
        loadImageFromUrl(holder.binding.projectImage, project.getIcon());
        loadImageFromUrl(holder.binding.screenshot1, project.getScreenshot1());
        loadImageFromUrl(holder.binding.screenshot2, project.getScreenshot2());
        loadImageFromUrl(holder.binding.screenshot3, project.getScreenshot3());

        holder.itemView.setScaleX(1f);
        holder.itemView.setScaleY(1f);

        holder.binding.getRoot().setOnClickListener(v -> {
            Intent intent = new Intent(Intent.ACTION_VIEW);
            intent.setData(Uri.parse("https://web.sketchub.in/p/" + project.getId()));
            holder.itemView.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        if (projects == null) {
            return 0;
        }
        return projects.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ViewStoreProjectPagerItemBinding binding;

        public ViewHolder(ViewStoreProjectPagerItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/besome/sketch/fragments/projects_store/adapters/StorePagerProjectsAdapter.kt
package com.besome.sketch.fragments.projects_store.adapters

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.besome.sketch.fragments.projects_store.ProjectPreviewActivity
import com.besome.sketch.fragments.projects_store.api.ProjectModel
import com.google.gson.Gson
import com.sketchware.remod.databinding.ViewStoreProjectPagerItemBinding
import mod.ilyasse.utils.UI

class StorePagerProjectsAdapter(
    private val projects: List<ProjectModel.Project>?,
    private val context: FragmentActivity
) :
    RecyclerView.Adapter<StorePagerProjectsAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ViewStoreProjectPagerItemBinding.inflate(inflater, parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val project = projects!![position]

        holder.binding.projectTitle.text = project.title
        holder.binding.projectDesc.text = project.description
        UI.loadImageFromUrl(holder.binding.projectImage, project.icon)
        UI.loadImageFromUrl(holder.binding.screenshot1, project.screenshot1)
        UI.loadImageFromUrl(holder.binding.screenshot2, project.screenshot2)
        UI.loadImageFromUrl(holder.binding.screenshot3, project.screenshot3)

        holder.itemView.scaleX = 1f
        holder.itemView.scaleY = 1f

        holder.binding.root.setOnClickListener { v: View? -> openProject(project) }
    }

    override fun getItemCount(): Int {
        if (projects == null) {
            return 0
        }
        return projects.size
    }

    class ViewHolder(val binding: ViewStoreProjectPagerItemBinding) :
        RecyclerView.ViewHolder(binding.root)

    private val gson = Gson()

    private fun openProject(project: ProjectModel.Project) {
        val fm = context.supportFragmentManager

        if (fm.findFragmentByTag("project_preview") == null) {
            val bundle = Bundle()
            bundle.putString("project_json", gson.toJson(project))

            val intent = Intent(context, ProjectPreviewActivity::class.java)
            intent.putExtras(bundle)
            context.startActivity(intent)
        }
    }
}
